src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -109,15 +118,42 @@ public override bool Execute()
             var logger = new MSBuildLoggerAdapter(Log);
             var config = TaskUtilities.TryGetConfig(AnalysisConfigDir, logger);
 
-            if (ShouldMergeAnalysisSettings(Language, config, logger))
+            var languageSettings = GetLanguageSpecificSettings(config);
+            if (languageSettings == null)
```

Comment:
All of the code paths need to find the language settings so this is now done up-front, rather than in multiple places.